src/commands/execute.ts

```diff
@@ -0,0 +1,199 @@
+import type { ProfileDocumentNode, ProviderJson } from '@superfaceai/ast';
+import {
+  AssertionError,
+  assertProviderJson,
+  isValidProviderName,
+} from '@superfaceai/ast';
+import { parseDocumentId, parseProfile, Source } from '@superfaceai/parser';
+
+import type { ILogger } from '../common';
+import type { Flags } from '../common/command.abstract';
+import { Command } from '../common/command.abstract';
+import type { UserError } from '../common/error';
+import {
+  buildMapPath,
+  buildProfilePath,
+  buildProviderPath,
+  buildRunFilePath,
+} from '../common/file-structure';
+import { exists, readFile } from '../common/io';
+import { execute } from '../logic/execution';
+
+export default class Execute extends Command {
+  // TODO: add description
+  public static description =
+    'This commands executes created integration. Commands `prepare`, `new` and `map` must be run before this command.';
+
+  public static examples = [
+    'superface execute <provider-name> <optional-profile-scope>.<profile-name> <optional-language>',
+  ];
+
+  public static args = [
+    {
+      name: 'providerName',
+      description: 'Name of provider.',
+      required: true,
+    },
+    {
+      name: 'profileId',
+      description: 'Id of profile, eg: starwars.character-information',
+      required: true,
+    },
+    {
+      name: 'language',
+      description: 'Language of generated integration. Default is JS.',
+      required: false,
+      default: 'JS',
+    },
+  ];
+
+  public static flags = {
+    ...Command.flags,
+  };
+
+  public async run(): Promise<void> {
+    const { flags, args } = this.parse(Execute);
+    await super.initialize(flags);
+    await this.execute({
+      logger: this.logger,
+      userError: this.userError,
+      flags,
+      args,
+    });
+  }
+
+  public async execute({
+    logger,
+    userError,
+    // flags,
+    args,
+  }: {
+    logger: ILogger;
+    userError: UserError;
+    flags: Flags<typeof Execute.flags>;
+    args: { providerName?: string; profileId?: string; language?: string };
+  }): Promise<void> {
+    const { providerName, profileId, language } = args;
+
+    // TODO: resuse check from New command
+    // Check provider name
+    if (providerName === undefined) {
```

Comment:
question: Is this check required? The `providerName` arg is required in the oclif command definition.

Reply:
Technically it isn't but the way we now extend Oclif base Command class is making all the arguments optional. I would like to fix that but I would need to change a lot of files - not now.